adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -1053,6 +1059,35 @@ public AdyenPaymentService getAdyenPaymentService() {
         return adyenPaymentServiceFactory.createFromBaseStore(baseStoreService.getCurrentBaseStore());
     }
 
+    /**
+     * Ininiate POS Payment using Adyen Terminal API
+     */
+    @Override
+    public OrderData initiatePosPayment(CartData cartData) throws Exception {
+        CustomerModel customer = null;
+        if (!getCheckoutCustomerStrategy().isAnonymousCheckout()) {
+            customer = getCheckoutCustomerStrategy().getCurrentUserForCheckout();
+        }
+
+        TerminalAPIResponse terminalApiResponse = getAdyenPaymentService().sendSyncPosPaymentRequest(cartData, customer);
+        ResultType resultType = getResultType(terminalApiResponse);
+
+        if (ResultType.SUCCESS == resultType) {
+            PaymentsResponse paymentsResponse = getPosPaymentResponseConverter().convert(terminalApiResponse.getSaleToPOIResponse());
+            return createAuthorizedOrder(paymentsResponse);
+        }
+
```

Comment:
We need to add all possible response codes here too. For example REFUSED is missing.

Reply:
Refactored to handle the FAILURE case and the cases when SaleToPOIRequest is present on the response.